### ISSUE TITLE
Support new Rails 5 syntax

### ISSUE THREAD

**Issue by leppert** [arthurnn/minitest-emacs#34]
Beginning with Rails 5.0.0, testing [is expected](http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#use-bin-rails-for-running-tasks-and-tests) to be run via the `rails test` command rather than `rake test`, with `rake test` being [fully deprecated](https://github.com/rails/rails/issues/22232#issuecomment-155456681) at some point in the future. The effect of this today is that paths and test names passed via `rake` appear to be ignored, instead running the entire suite.

Not sure if the best solution here is to add a new `(defcustom use-rails-5-syntax)` or try to infer the appropriate syntax based on the project.

**Comment by arthurnn** [comment]
would be awesome to infer it from the project, not sure how we would be able to do that tho

**Comment by leppert** [comment]
`rails -v` gives you the version, which you could then test against. And `rails -h` gives you all the commands to which `rails` responds, which could be used to see if "test" is one of the supported commands.

The main concern I have with both of these is that they're slow—too slow to run before each test run. But they might be fine for setting some sort of project level config, something like `minitest-infer-and-write-config` that then sets defs in `.dir-locals.el`. Awkward, and would need to be rerun if you upgrade or downgrade Rails, but maybe not so bad? Wonder if there are any other Emacs libs that do something similar.

**Comment by arthurnn** [comment]
@leppert interesting. we need to keep in mind this should also work outside rails, for ruby only projects.
But I guess what you said make sense 👍

**Comment by arronmabrey** [comment]
Just bumped into this as well using minitest-emacs in spacemacs. Following along. :-)

**Comment by Linuus** [comment]
Setting `minitest-use-rails` to `t` will use the `rails test ...` command.

**Comment by gcentauri** [comment]
> Setting `minitest-use-rails` to `t` will use the `rails test ...` command.

Indeed. This config is newer than this issue, and solves the original problem. I still think it could be nice to auto-detect a rails project and version to see if `rake` or `bin/rails` is preferred.
